### PR TITLE
Next & Last sessions

### PR DESCRIPTION
**Pull request by oliveiraigorm**

### CODE REVIEW

File: lib/Controller/PageController.php [hunk 1]
```diff
@@ -252,4 +255,146 @@ public function getClientSessions(int $clientId): array {
...
 		return $sessions;
 	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * Get next session for a specific client
+	 */
+
```
[the remark this answers is not in the document]
Comment (oliveiraigorm) [reply lib/Controller/PageController.php:265]: Did you mean shouldn't? 
Should it be like this then?
image

[the remark this answers is not in the document]
Comment (mwalbeck) [reply lib/Controller/PageController.php:265]: Ahh yeah, sorry about that ...

Yes, there shouldn't be an empty line

File: lib/Controller/PageController.php [hunk 2]
```diff
@@ -252,4 +255,149 @@ public function getClientSessions(int $clientId): array {
...
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * Get last session for a specific client
+	 */
+	public function getClientLastSession(int $clientId) {
+		$dateNow = new DateTime();
+
+		$sixWeeksAgo = new DateTime();
+		$sixWeeksAgo = $sixWeeksAgo->modify("-6 week");
```
Comment (frozenbanana) [inline lib/Controller/PageController.php:340]: Why is this needed?

Comment (frozenbanana) [reply lib/Controller/PageController.php:340]: Or more precisely. Are we using the magic time distance of 6 week back in time to find last session?

So if the last session was 6 weeks and one day ago there was no last session?

Comment (frozenbanana) [reply lib/Controller/PageController.php:340]:
```suggestion
		$userStartTime = $this->getTimeUserCreated($this->userId);
```

File: src/components/Table.vue [hunk 3]
```diff
@@ -126,6 +131,9 @@ export default {
 		timezoneWithUTC(timezone) {
 			return TimezoneUtil.timezoneWithUTC(timezone);
 		},
+		formatDate(date) {
+			if (date) return dayjs(date).format("DD/MM hh:mm");
```
Comment (frozenbanana) [inline src/components/Table.vue:135]: We should think about locales and format our dates and times accordingly. @mwalbeck A topic to bring up on tech checkin meeting.

**Review by frozenbanana — CHANGES_REQUESTED** [review]
Check it out and talk to me if it makes sense.

File: src/components/Table.vue (hunk 3 above)
Comment (oliveiraigorm) [reply src/components/Table.vue:135]: Yes, definitely. I wasn't sure about which format to use and left it that way so it's easily modifiable.

**Review by mwalbeck — APPROVED** [review]

**Review by mwalbeck — APPROVED** [review]

**Pull request merged**